tests/test_core.py

```diff
@@ -618,6 +618,35 @@ def test_path_check_invalid_http_method(self, spec):
         with pytest.raises(APISpecError, match=message):
             spec.path("/pet/{petId}", operations={"dummy": {}})
 
+    def test_path_resolve_response_schema(self, spec):
+        schema = {"schema": "PetSchema"}
```

Comment:
This kind of input isn't comparable with using the Marshmallow plugin, right? From what I'm seeing the Marshmallow plugin will get called first look to resolve a  Marshmallow schema for the string "PetSchema" and then raise an exception. I might be wrong about that and I haven't tested it though.

Reply:
You're right, indeed. This is not compatible.

We could leave it as is and decide that the app should be either all-marshmallow or no-marshmallow and support this only if marshmallow plugin is not used.

Or we adapt the marshmallow plugin to pass strings, but this means it won't validate that the string corresponds to an existing schema. The schema instance resolver won't raise on unknown schema but return a string if the schema is unknown. We may hide typos or import order issues, but that's life.

It looks feasible. I didn't think it through but I'm not opposed to the idea.

Should this be considered a breaking change?

Reply:
We could even merge this is a non-breaking fix for users not using the marshmallow plugin and fix the marshmallow plugin compatibility later.